src/tric.php

```diff
@@ -833,21 +846,77 @@ function update_stack_images() {
 }
 
 /**
- * Maybe runs composer install on a given target
+ * Check if a recognized command's required file exists in the specified directory.
  *
- * @param array $base_command Base command to run.
- * @param string $target Target to potentially run composer install against.
- * @param array $sub_directories Sub directories to prompt for additional execution.
+ * @param string $base_command Command name, such as 'composer' or 'npm'.
+ * @param string $path         The directory path in which to look for relevantly-required files (e.g. 'package.json').
  *
- * @return null|int Result of command execution.
+ * @return bool True if the path is a directory and the command doesn't have a known file requirement or the expected
+ *              file does exist. False if the path is not a directory or a recognized command didn't find the
+ *              relevantly-required file.
+ */
+function dir_has_req_build_file( $base_command, $path ) {
+	// Bail if doesn't exist or is not a directory.
+	if (
+		! file_exists( $path )
```

Comment:
```suggestion
		! is_file( $path )
```

`is_file` caches the result internally and checks for permissions too, usually better than using `file_exists`.